js/containers/ApprovalRO.jsx

```diff
@@ -0,0 +1,160 @@
+				return BYPASSED
+			}
+			return APPROVED;
+		default:
+			return '';
+	}
+}
+
+function getStateTitle(approvalState) {
+	switch (approvalState) {
+		case PENDING:
+			return 'Pending review by:';
+		case REJECTED:
+			return 'Rejected by:';
+		case APPROVED:
+			return 'Approved by:';
+		case BYPASSED:
+			return 'Approval have been bypassed'
+		default:
+			return '';
+	}
+}
+
+function getStateDescription(approvalState) {
+	switch (approvalState) {
+		case PENDING:
+			return 'Send a request to deploy this release, once approved team members will have the ability to deploy this release. Only one request for approval can be sent at a time, although approval can also be granted by others with the same permission.';
+		case REJECTED:
+			return 'Send a request to deploy this release, once approved team members will have the ability to deploy this release. Only one request for approval can be sent at a time, although approval can also be granted by others with the same permission.';
+		case BYPASSED:
+		case APPROVED:
+			return 'Once approved team members will have the ability to deploy this release. Only one request for approval can be sent at a time, although approval can also be granted by others with the same permissions. e.g. Release managers.';
+		default:
+			return '';
+	}
+}
+
+function Approval(props) {
+	let error = null;
+	if (props.error) {
+		error = (
+			<div className="alert alert-danger">
+				<div className="">
+					{props.error}
+				</div>
+			</div>
+		);
+	}
+
+	const approvalState = getApprovalState(props.deployment_state)
+
+	let stateTitle = getStateTitle(approvalState);
+	let approver_name = props.approver ? props.approver.name : '';
+
+	let date = '';
+	if (props.date_approved_nice) {
+		date = props.date_approved_nice;
+	} else if (props.date_requested_nice) {
+		date = props.date_requested_nice;
+	}
+
+	if (approver_name === '' && approvalState === APPROVED) {
+		stateTitle = 'Approval have been bypassed';
+	}
+
+	let action = null;
+	if (approvalState == PENDING) {
+		action = (
+			<a className="approval-action" onClick={props.onCancel}>
+				<i className="fa fa-times-circle"></i> Remove
+			</a>
+		);
+	}
+
+	return (
+		<div className="section approval">
+			<header id="2">Approval</header>
+			<p>
+				{getStateDescription(approvalState)}
+			</p>
+			<StatusBox type={approvalState}>
+				<div>
+					{action}
+					<div className={"state " + approvalState}>
+						{stateTitle}
+					</div>
+				</div>
+				<div>{approver_name}
+					<small>{date}</small>
+				</div>
+			</StatusBox>
+			<div>
+				<ApproveRequest /> <RejectRequest />
+			</div>
+			<div>
+				<Bypass />
+			</div>
+			{error}
+		</div>
+	);
+}
+
+const mapStateToProps = function(state) {
+	let approver = state.deployment.approvers.find(function(val) {
+		if (val.id === state.deployment.approver_id) {
+			return val
+		}
+	});
+
+	return {
+		deployment_state: state.deployment.list[state.deployment.id].state.toLowerCase(),
```

Comment:
can't you get this from `state.deployment.data.state`? Also, why go to lowercase?